CI bug: default go version installed via `apt` cannot satisfy the requires from editorconfig-core-go

[CI lint report](https://github.com/tj/git-extras/actions/runs/6506122881/job/17671061777?pr=1084#step:4:117)

```
Setting up golang-1.18-doc (1.18.1-1ubuntu1.1) ...
Setting up golang-1.18-src (1.18.1-1ubuntu1.1) ...
Setting up golang-src (2:1.18~0ubuntu2) ...
Setting up golang-1.18-go (1.18.1-1ubuntu1.1) ...
Setting up golang-1.18 (1.18.1-1ubuntu1.1) ...
Setting up golang-go:amd64 (2:1.18~0ubuntu2) ...
Setting up golang-doc (2:1.18~0ubuntu2) ...
Setting up golang:amd64 (2:1.18~0ubuntu2) ...
Processing triggers for man-db (2.10.2-1) ...
NEEDRESTART-VER: 3.5
NEEDRESTART-KCUR: 6.2.0-1012-azure
NEEDRESTART-KEXP: 6.2.0-1012-azure
NEEDRESTART-KSTA: 1
go: downloading github.com/editorconfig-checker/editorconfig-checker v0.0.0-20231013082012-98f0d6560dba
go: downloading github.com/editorconfig/editorconfig-core-go/v2 v2.6.0
go: downloading github.com/gabriel-vasile/mimetype v1.4.3
go: downloading golang.org/x/mod v0.12.0
go: downloading gopkg.in/ini.v1 v1.67.0
go: downloading github.com/baulk/chardet v0.1.0
go: downloading golang.org/x/text v0.13.0
go: downloading golang.org/x/net v0.17.0
# github.com/editorconfig/editorconfig-core-go/v2
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/cached_parser.go:63:21: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/config.go:29:16: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/config.go:76:21: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/definition.go:48:20: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/definition.go:61:20: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/definition.go:75:20: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/editorconfig.go:[92](https://github.com/tj/git-extras/actions/runs/6506122881/job/17671061777?pr=1084#step:4:93):21: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/editorconfig.go:214:16: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/editorconfig.go:241:16: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/editorconfig.go:258:16: undefined: errors.Join
../../../go/pkg/mod/github.com/editorconfig/editorconfig-core-go/v2@v2.6.0/editorconfig.go:258:16: too many errors
note: module requires Go 1.20
```

No locked file to match the wished version for editorconfig-core-go, and the default go cannot satisfy the requires.

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -17,9 +17,12 @@ jobs:
         # reviewing!
       - name: Check out code.
         uses: actions/checkout@v3
+      - uses: 'actions/setup-go@v4'
+        with:
+          go-version: '1.20'
       - name: 'Check EditorConfig Lint'
         run: |
-          sudo apt install -y jq golang
+          sudo apt install -y jq
           go install 'github.com/editorconfig-checker/editorconfig-checker/cmd/editorconfig-checker@latest'
 
           readarray -t changed_files <<<"$(jq -r '.[]' <<<'${{ steps.files.outputs.added_modified }}')"

